src/main/java/org/terasology/climateConditions/HypothermiaSystem.java

```diff
@@ -0,0 +1,115 @@
+    public static final String FROSTBITE_DAMAGE_ACTION_ID = "Frostbite Damage";
+
+    @In
+    private DelayManager delayManager;
+    @In
+    private PrefabManager prefabManager;
+    @In
+    private Context context;
+
+    private float thresholdHeight = 60f;
+    private int healthDecreaseInterval = 20000;
+    private int initialDelay = 5000;
+    private int healthDecreaseAmount = 15;
+    private Random random = new FastRandom();
+
+
+    @ReceiveEvent(components = {PlayerCharacterComponent.class, CharacterMovementComponent.class})
+    public void observeDangerZone(MovedEvent event, EntityRef player, LocationComponent location, CharacterMovementComponent movement) {
+        //TODO: react on OnBiomeChangedEvent to handle the danger zone
+        float height = event.getPosition().getY();
+        float lastHeight = height - event.getDelta().getY();
+        if (height > thresholdHeight && lastHeight <= thresholdHeight) {
+            player.addOrSaveComponent(new HypothermiaComponent());
+        }
+        if (height < thresholdHeight && lastHeight >= thresholdHeight) {
+            if(player.hasComponent(HypothermiaComponent.class)){
+                player.removeComponent(HypothermiaComponent.class);
+            }
+        }
+    }
+
+    @ReceiveEvent
+    public void onHypothermia(OnAddedComponent event, EntityRef player, HypothermiaComponent hypothermia){
+        if(player.hasComponent(HypothermiaComponent.class)){
+            delayManager.addPeriodicAction(player, FROSTBITE_DAMAGE_ACTION_ID, initialDelay, healthDecreaseInterval);
+        }
+    }
+    @ReceiveEvent(components = {LocationComponent.class})
+    public void onPeriodicFrostbite(PeriodicActionTriggeredEvent event, EntityRef player) {
+        if (event.getActionId().equals(FROSTBITE_DAMAGE_ACTION_ID)) {
+            // Check to see if health should be decreased
+            LocationComponent location = player.getComponent(LocationComponent.class);
+            float height = location.getLocalPosition().getY();
+            if (height > thresholdHeight) {
+                applyFrostbiteDamagePlayer(player);
+                // Stun the player for 500 ms
+                applyStunEffect(player, 1000);
+                playFrostbiteSound(player);
+            }
+        }
+    }
+    private void applyFrostbiteDamagePlayer(EntityRef player) {
+        Prefab frostbiteDamagePrefab = prefabManager.getPrefab("ClimateConditions:FrostbiteDamage");
+        player.send(new DoDamageEvent(healthDecreaseAmount, frostbiteDamagePrefab));
+    }
+    private void applyStunEffect(EntityRef player, int duration) {
+        StunAlterationEffect stunAlterationEffect = new StunAlterationEffect(context);
+        //Both the instigator and the target is the player
+        //the magnitude parameter is not used by StunAlterationEffect
+        stunAlterationEffect.applyEffect(player, player, 0, duration);
+
+    }
+    public void playFrostbiteSound (EntityRef entity) {
```

Comment:
```suggestion
    public void playFrostbiteSound(EntityRef entity) {
```
Intellij code optimization (Ctrl+Alt+L) is your friend :wink: